exchange/bidder.go

```diff
@@ -176,19 +176,25 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, request *openrtb2.B
 		httpInfo := <-responseChannel
 		// If this is a test bid, capture debugging info from the requests.
 		// Write debug data to ext in case if:
+		// - headerDebugAllowed (debug override header specified correct) - it overrides all other debug restrictions
 		// - debugContextKey (url param) in true
 		// - account debug is allowed
 		// - bidder debug is allowed
-		if debugInfo := ctx.Value(DebugContextKey); debugInfo != nil && debugInfo.(bool) {
-			if accountDebugAllowed {
-				if bidder.config.DebugInfo.Allow {
-					seatBid.httpCalls = append(seatBid.httpCalls, makeExt(httpInfo))
-				} else {
-					debugDisabledWarning := errortypes.Warning{
-						WarningCode: errortypes.BidderLevelDebugDisabledWarningCode,
-						Message:     "debug turned off for bidder",
+		if headerDebugAllowed {
+			seatBid.httpCalls = append(seatBid.httpCalls, makeExt(httpInfo))
```

Comment:
Please consider asserting that this information exists when `headerDebugAllowed` is set to `true` in one of the `requestBid` test cases

Reply:
Mansi, this is sort of done in test `TestRequestBidRemovesSensitiveHeaders`, line 183. 
It works because `accountDebugAllowed` for this test case is set to true. In this case it works similarly to the case where `headerDebugAllowed` is set to true. There are more cases to test debug options combination and results in `TestDebugBehavior` in exchange_test.go file.
Do you think I need to add additional test (or modify existing one) in `requestBid` test cases?

Reply:
@VeronikaSolovei9 Thanks for that explanation. However, even though both `accountDebugAllowed` and `headerDebugAllowed` kind of have the same purpose in this function, they are both different flags and so it would be a good idea to assert the functionality based on explicit values of both the flags